### DISCUSSION TITLE
How to add a custom language to be used with mkdocs?

### DISCUSSION THREAD

**Discussion by mattias-mohlin** [PrismJS/prism#3482] [Q&A] [answered]
I've tried to follow the steps at https://prismjs.com/extending.html#creating-a-new-language-definition for adding a new language definition to Prism. Everything works fine on the test.html page and I can see the keywords of my language colorized. However, how should I proceed from here to actually getting a prism.js file that contains my new language which I can include in my mkdocs site?
I also notice something strange when opening the download.html file. My language appears in the list, but with a size of 0 KB.
I think I'm missing something obvious, and I'm pretty sure that if I just can get the correct prism.js file generated then it will work with mkdocs because it works for a built-in language like C++.

**Comment by RunDevelopment** [comment, marked as answer]
To get the download page working locally, you have to run it with a local HTTP server (e.g. [this one](https://www.npmjs.com/package/http-server)). Then the download page will work.

Also, be sure to run `npm run build` before "downloading" the generated bundle. The download page load `components.js`, so that file has to be rebuilt.

> My language appears in the list, but with a size of 0 KB

That's because we get the sizes from our GitHub repo. Since your languages isn't included, it will be displayed as 0KB. 

However, this doesn't affect the bundle. It's just user-facing information.

**Reply by mattias-mohlin** [reply]
Thanks, with a local HTTP server it works fine!
